components/google-drive/google-drive.app.js

```diff
@@ -0,0 +1,159 @@
+const axios = require("axios");
+const { google } = require("googleapis");
+
+const GOOGLE_DRIVE_UPDATE_TYPES = [
+  "sync",
+  "remove",
+  "update",
+  "trash",
+  "untrash",
+  "change",
+];
+
+module.exports = {
+  type: "app",
+  app: "google_drive",
+  propDefinitions: {
+    files: {
+      type: "string[]",
+      label: "Files",
+      description: "The files you want to watch for changes.",
+      optional: true,
+      async options({ page, prevContext }) {
+        const { nextPageToken } = prevContext;
+        return await this.listFiles(nextPageToken);
+      },
+    },
+    updateTypes: {
+      type: "string[]",
+      label: "Types of updates",
+      description:
+        "The types of updates you want to watch for on these files. [See Google's docs](https://developers.google.com/drive/api/v3/push#understanding-drive-api-notification-events).",
+      // Everything from this doc except for "add", which is irrelevant
+      // since the files already exists:
+      // https://developers.google.com/drive/api/v3/push#understanding-drive-api-notification-events
+      default: GOOGLE_DRIVE_UPDATE_TYPES,
+      options: GOOGLE_DRIVE_UPDATE_TYPES,
+    },
+    watchForPropertiesChanges: {
+      type: "boolean",
+      label: "Watch for changes to file properties",
+      description:
+        "Watch for changes to [file properties](https://developers.google.com/drive/api/v3/properties) in addition to changes to content. **Defaults to `false`, watching for only changes to content**.",
+      optional: true,
+      default: false,
+    },
+  },
+  methods: {
+    // Returns a drive object authenticated with the user's access token
+    drive() {
+      const auth = new google.auth.OAuth2();
+      auth.setCredentials({ access_token: this.$auth.oauth_access_token });
+      const drive = google.drive({ version: "v3", auth });
+      return drive;
+    },
+    // Google's push notifications provide a URL to the resource that changed,
+    // which we can use to fetch the file's metadata. So we use axios here
+    // (vs. the Node client) to get that.
+    async getFileMetadata(url) {
```

Comment:
ok this looks like it will either succeed or throw so we gucci (*should* have no null/undefined issue)